fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintainer.java

```diff
@@ -264,6 +267,19 @@ private Map<IndexOptions, List<LuceneDocumentFromRecord.DocumentField>> getIndex
     @SuppressWarnings("PMD.CloseResource")
     private void deleteDocument(Tuple groupingKey, Tuple primaryKey) throws IOException {
         final IndexWriter oldWriter = directoryManager.getIndexWriter(groupingKey, indexAnalyzerSelector.provideIndexAnalyzer(""));
+        final DirectoryReader directoryReader = DirectoryReader.open(oldWriter);
+        @Nullable final LucenePrimaryKeySegmentIndex segmentIndex = directoryManager.getDirectory(groupingKey).getPrimaryKeySegmentIndex();
+        if (segmentIndex != null) {
+            final LucenePrimaryKeySegmentIndex.DocumentIndexEntry documentIndexEntry = segmentIndex.findDocument(directoryReader, primaryKey);
+            if (documentIndexEntry != null) {
```

Comment:
If the document to be deleted's segment is in the writer's `runningMerges`, then that segment will not appear in the NRT reader from that writer. (Also the `.cfs` file won't exist yet; the stored fields will still be in`.f*` files. Although that doesn't matter a lot.) So `segmentIndex.findDocument` will find the index entry not it will look like an orphan and `documentIndexEntry` will be `null`.